dali/base/dadfs.cpp

```diff
@@ -586,6 +587,57 @@ class CFileSuperOwnerLock : protected CFileLockCompound
     {
         return CFileLockCompound::detach();
     }
+    bool initWithFileLock(const CDfsLogicalFileName &logicalName, IRemoteConnection *conn, unsigned timeout, const char *msg, CFileLock &fcl, unsigned fclmode)
+    {
+        // SuperOwnerLock while holding fcl
+        IRemoteConnection *fclConn = fcl.queryConnection();
+        CTimeMon tm(timeout);
+        unsigned remaining = timeout;
+        unsigned interval = 5000;
+        loop
+        {
+            if (tm.timedout(&remaining))
+                return false;
+            try
+            {
+                if (init(logicalName, conn, 0, msg))
+                    return true;
+            }
+            catch (ISDSException *e)
+            {
+                if (SDSExcpt_LockTimeout != e->errorCode())
+                    throw;
+                e->Release();
+            }
+            if (tm.timedout(&remaining))
+                return false;
+            unsigned delta = remaining;
+            if (delta < interval)
+                delta = interval;
+            // release lock
+            {
+                CriticalBlock block(superOwnerLockSect);
+                fclConn->changeMode(RTM_NONE, delta);
+                fclConn->reload();
```

Comment:
You don't need reload() here.. it's only after regaining lock and using the connection that a forced reload() is needed.
Whilst sleeping it can remain stale.